libultraship/libultraship/CMakeLists.txt

```diff
@@ -0,0 +1,516 @@
+    "Factories/ResourceLoader.h"
+    "Material.h"
+    "Matrix.h"
+    "Model.h"
+    "Path.h"
+    "PlayerAnimation.h"
+    "Scene.h"
+    "Skeleton.h"
+    "SkeletonLimb.h"
+    "Text.h"
+    "Texture.h"
+    "Vertex.h"
+)
+source_group("Header Files\\Resources\\Files" FILES ${Header_Files__Resources__Files})
+
+set(Source_Files__Audio
+    "abi.h"
+    "AudioPlayer.h"
+    "mixer.c"
+    "mixer.h"
+    "SDLAudioPlayer.cpp"
+    "SDLAudioPlayer.h"
+)
+
+if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
+set(Source_Files__Audio__Windows
+    "WasapiAudioPlayer.cpp"
+    "WasapiAudioPlayer.h"
+)
+else()
+set(Source_Files__Audio__Linux
+    "PulseAudioPlayer.cpp"
+    "PulseAudioPlayer.h"
+)
+endif ()
+
+source_group("Source Files\\Audio" FILES ${Source_Files__Audio} ${Source_Files__Audio__Windows} ${Source_Files__Audio__Linux})
+
+set(Source_Files__Config
+    "ConfigFile.cpp"
+    "ConfigFile.h"
+)
+source_group("Source Files\\Config" FILES ${Source_Files__Config})
+
+set(Source_Files__Controller
+    "Controller.cpp"
+    "Controller.h"
+    "KeyboardController.cpp"
+    "KeyboardController.h"
+    "SDLController.cpp"
+    "SDLController.h"
+    "UltraController.h"
+)
+source_group("Source Files\\Controller" FILES ${Source_Files__Controller})
+
+set(Source_Files__Controller__Attachment
+    "ControllerAttachment.cpp"
+    "ControllerAttachment.h"
+    "MemoryPack.cpp"
+    "MemoryPack.h"
+    "RumblePack.cpp"
+    "RumblePack.h"
+)
+source_group("Source Files\\Controller\\Attachment" FILES ${Source_Files__Controller__Attachment})
+
+set(Source_Files__CustomImpl
+    "GameOverlay.cpp"
+    "GameOverlay.h"
+    "GameSettings.cpp"
+    "GameSettings.h"
+    "SohConsole.cpp"
+    "SohConsole.h"
+    "SohImGuiImpl.cpp"
+    "SohImGuiImpl.h"
+)
+source_group("Source Files\\CustomImpl" FILES ${Source_Files__CustomImpl})
+
+set(Source_Files__CustomImpl__Environment
+    "Environment.cpp"
+    "Environment.h"
+)
+source_group("Source Files\\CustomImpl\\Environment" FILES ${Source_Files__CustomImpl__Environment})
+
+set(Source_Files__CustomImpl__Hooks
+    "SohHooks.cpp"
+    "SohHooks.h"
+)
+source_group("Source Files\\CustomImpl\\Hooks" FILES ${Source_Files__CustomImpl__Hooks})
+
+set(Source_Files__CustomImpl__Utils
+    "Utils.cpp"
+    "Utils.h"
+)
+source_group("Source Files\\CustomImpl\\Utils" FILES ${Source_Files__CustomImpl__Utils})
+
+set(Source_Files__Globals
+    "Cvar.cpp"
+    "Cvar.h"
+    "GlobalCtx2.cpp"
+    "GlobalCtx2.h"
+    "LUSMacros.h"
+    "Window.cpp"
+    "Window.h"
+    "WindowShim.cpp"
+    "WindowShim.h"
+)
+source_group("Source Files\\Globals" FILES ${Source_Files__Globals})
+
+set(Source_Files__Lib
+    "Lib/mINI/src/mini/ini.h"
+    "Lib/StrHash64.cpp"
+    "Lib/StrHash64.h"
+    "Lib/tinyxml2/tinyxml2.cpp"
+    "stox.cpp"
+    "stox.h"
+)
+source_group("Source Files\\Lib" FILES ${Source_Files__Lib})
+
+set(Source_Files__Lib__Fast3D
+    "Lib/Fast3D/gfx_cc.cpp"
+    "Lib/Fast3D/gfx_cc.h"
+    "Lib/Fast3D/gfx_direct3d11.cpp"

```

Comment:
we probably only want to include the relevant ones per platform right? D3D only on windows and glx only on LInux?